Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Extensions/FbDbContextOptionsExtensions.cs

```diff
@@ -28,15 +28,15 @@ public static class FbDbContextOptionsExtensions
 	{
 		public static DbContextOptionsBuilder UseFirebird(this DbContextOptionsBuilder optionsBuilder, string connectionString, Action<FbDbContextOptionsBuilder> fbOptionsAction = null)
 		{
-			var extension = GetOrCreateExtension(optionsBuilder).WithConnectionString(connectionString);
+			var extension = (FbOptionsExtension)GetOrCreateExtension(optionsBuilder).WithConnectionString(connectionString);
 			((IDbContextOptionsBuilderInfrastructure)optionsBuilder).AddOrUpdateExtension(extension);
 			fbOptionsAction?.Invoke(new FbDbContextOptionsBuilder(optionsBuilder));
 			return optionsBuilder;
 		}
 
 		public static DbContextOptionsBuilder UseFirebird(this DbContextOptionsBuilder optionsBuilder, DbConnection connection, Action<FbDbContextOptionsBuilder> fbOptionsAction = null)
 		{
-			var extension = GetOrCreateExtension(optionsBuilder).WithConnection(connection);
+			var extension = (FbOptionsExtension)GetOrCreateExtension(optionsBuilder).WithConnection(connection);
```

Comment:
What's the reason for the casting?

Reply:
You will not be able to recover the connection properties for example!

With this -> options.FindExtension() in FbOptions

And I already tested !!!

Reply:
As I explained, we can use this CAST!